File: 11687.py
M = int(input())

def cnt(M, cnt):
	if cnt == M:
		print(cnt*5)
		return
	elif cnt > M:
		print(-1)
		return
	else:
		tmp = M
		while tmp > 0:
			cnt += 1
			tmp /= 5

cnt(M, 1)

File: 11687.cpp
#include <iostream>

using namespace std;

int M,cnt;

int main(void) {
	cin >> M;

	int left=1,right=M*5;
	while (left <= right) {
		int mid = (right+left)/2;

		cnt=0;
		for (int i=5; i<=mid; i*=5) {
			cnt += mid / i;
		}

		if (cnt >= M) {
			right = mid-1;
		}
		else {
			left = mid+1;
		}
	}

	cnt=0;
	for (int i=5; i<=left; i*=5) {
		cnt += left / i;
	}

	if (cnt == M) {
		cout << left << '\n';
	}
	else {
		cout << -1 << '\n';
	}
	return 0;
}